pkg/filter/dt/filter_mysql_query.go

```diff
@@ -166,17 +176,19 @@ func (f *_mysqlFilter) processAfterQueryUpdate(ctx context.Context, conn *driver
 	log.Debugf("update, lockKey: %s", lockKeys)
 	undoLog := exec.BuildUndoItem(false, constant.SQLType_UPDATE, schemaName, executor.GetTableName(), lockKeys, beforeImage, afterImage)
 
-	branchID, err := f.registerBranchTransaction(ctx, xid, conn.DataSourceName(), lockKeys)
+	branchID, err := f.registerBranchTransaction(newCtx, xid, conn.DataSourceName(), lockKeys)
 	if err != nil {
 		return err
 	}
 	log.Debugf("update, branch id: %d", branchID)
 	return dt.GetUndoLogManager().InsertUndoLogWithNormal(conn, xid, branchID, undoLog)
 }
 
-func (f *_mysqlFilter) processSelectForQueryUpdate(ctx context.Context, conn *driver.BackendConnection,
-	result proto.Result, selectStmt *ast.SelectStmt) error {
+func (f *_mysqlFilter) processSelectForQueryUpdate(
+	ctx context.Context, conn *driver.BackendConnection, result proto.Result, selectStmt *ast.SelectStmt) error {
 	has, _ := misc.HasXIDHint(selectStmt.TableHints)
+	_, span := tracing.GetTraceSpan(ctx, "mysql_process_select_for_query_update")
```

Comment:
I recommend not adding TraceContext for this location. instead, add inside of func `executor.Executable`:
```
newCtx, span := tracing.GetTraceSpan(ctx, "executable")
```